### PR TITLE
fix(report): Result filter issue and async tab ordering issue

### PR DESCRIPTION
**Pull request by jbrundage**
Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

### CODE REVIEW

**Comment by jbrundage** [comment]
@GordonSmith @dcamper please review

File: report/src/index.ts [hunk 1]
```diff
@@ -45,21 +45,30 @@ export class ReportTabs extends DockPanel {
...
                             let tabName = 'Report ' + i;
                             if (result.Name) {
                                 tabName = result.Name;
                             }
-                            if (i === 0) {
-                                this.addWidget(r, tabName);
-                            } else {
-                                this.addWidget(r, tabName, "tab-after", prevWidget);
+                            reportNameMap[i] = tabName;
+                            if(resultCount === resultLimit){
```
Comment (GordonSmith) [inline report/src/index.ts]: Should really use a `Promise.all` (code will get simpler):  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Your code will end up looking a but like:
```
Promise.all(resultsArr,map(r=>r.fetchRows())).then(results=> ...));
```

Comment (dcamper) [reply report/src/index.ts]: @jbrundage Bump

Comment (jbrundage) [reply report/src/index.ts]: Ah okay, I'll make this change and test it today.

Comment (jbrundage) [reply report/src/index.ts]: @GordonSmith I've made the requested change:
https://github.com/hpcc-systems/DataPatterns/blob/4a35ecb22f05b332528136d12c1f3896044a072c/report/src/index.ts#L52-L68

**Comment by jbrundage** [comment]
@GordonSmith these changes are completed and ready for review

**Pull request merged**